base/server/src/main/java/com/netscape/cmscore/usrgrp/UGSubsystem.java

```diff
@@ -698,6 +699,9 @@ public void addUser(User user) throws EUsrGrpException {
             attrs.add(attr);
         }
 
+        logger.debug("UGSubsystem: - nsPagedSizeLimit: " + LDAP_ATTR_NS_PAGED_SIZE_LIMIT);
+        attrs.add(new LDAPAttribute("nsPagedSizeLimit", LDAP_ATTR_NS_PAGED_SIZE_LIMIT));
+
```

Comment:
Actually the `nsPagedSizeLimit` attribute only needs to be added to an LDAP user that will be used to create LDAP connections. In IPA case that would be `pkidbuser`. However, for general PKI users (e.g. admins, agents) this attribute is not needed, so it should be optional. Also, in case we need to change the value later it will be challenging to update all users in LDAP, so I think we should limit that to `pkidbuser` only.

I'd suggest we change [this code](https://github.com/dogtagpki/pki/blob/master/base/server/python/pki/server/deployment/__init__.py#L2644-L2648) to specifically add the attribute to `pkidbuser` like this:
```
subsystem.add_user(
    'pkidbuser',
    full_name='pkidbuser',
    user_type='agentType',
    state='1',
    nsPagedSizeLimit=20000)
```
or this if you want to make it more reusable:
```
subsystem.add_user(
    'pkidbuser',
    full_name='pkidbuser',
    user_type='agentType',
    state='1',
    attributes={
        nsPagedSizeLimit: 20000
    })
```

Reply:
There's no code for `attributes` right, so I would need to add a new argument for `add_user` and have code in the Java side to handle that new field?

Reply:
That's correct.